src/System.Private.CoreLib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -42,7 +44,7 @@ internal sealed partial class TlsOverPerCoreLockedStacksArrayPool<T> : ArrayPool
         /// An array of per-core array stacks. The slots are lazily initialized to avoid creating
         /// lots of overhead for unused array sizes.
         /// </summary>
-        private readonly PerCoreLockedStacks[] _buckets = new PerCoreLockedStacks[NumBuckets];
+        private readonly PerCoreLockedStacks?[] _buckets = new PerCoreLockedStacks[NumBuckets];
         /// <summary>A per-thread array of arrays, to cache one array per array size per thread.</summary>
         [ThreadStatic]
         private static T[][] t_tlsBuckets;
```

Comment:
I was going to suggest this should be  `private static T[][]? tlsBuckets;`  (since although array entries get special treatment, the reference to the array itself is clearly initially null here) but it seems the compiler only requires a `?` here if it is non static..?

The does not warn:
```c#
#nullable enable

public class C
{
    public static object o;	        
       
    public void N()
    {
          o = new object();
    }
} 
```
this does warn
```c#
#nullable enable

public class C
{
    public object o;	        
       
    public void N()
    {
          o = new object();
    }
} 
```

Reply:
Yup, changed.  I'm not sure why the compiler doesn't warn about this.  @jaredpar?

Reply:
Opened https://github.com/dotnet/roslyn/issues/34668